### ISSUE TITLE
Multiple bots on one machine with docker-compose

### ISSUE THREAD

**Issue by guizmoau** [freqtrade/freqtrade#3645]
I am trying to figure out how to start another bot on the same machine when using docker-compose ?

I didn't find infos on the docs for that specific use
https://www.freqtrade.io/en/latest/advanced-setup/#running-multiple-instances-of-freqtrade

I understand that it need different db etc...but how to start another command in parallel from docker-compose ? or any other tricks to do so...

Anyone can help ?

**Comment by xmatthias** [comment]
you have 2 options

copy the compose file to another directory and start it from there (it's a completely seperate instance - so it'll by default use different mount volumes, different configs, different database, ...).


it would then look as follows:
```
bot1
| docker-compose.yml
| -- user_data
...
bot2
| docker-compose.yml
| -- user_data
...

```

it's cleanly seperated ... and works fine.
---

the alternative is to create a 2nd service within docker compose (best familiarize yourself / read up on the syntax of docker-compose files first) - then you use different mount volumes ... and you can controll all bot's with `docker-compose up -d` or `docker-compose down`.

obviously, you can also use `docker-compose up -d freqtrade2` ... to just start the 2nd one.

**Comment by guizmoau** [comment]
ok that is quite simple and easy !!

Thanks.

Shall we update the documentation ?

**Comment by xmatthias** [comment]
I'm not certain which way should be documented - as both have benefits / drawbacks - and which one is better will depend on the familiarity with docker itself (not everyone is comfortable to work with services - as this will complicate certain commands ...).

**Comment by guizmoau** [comment]
I run into an error with telegram when starting the second docker-compose 

telegram.error.Conflict: Conflict: terminated by other getUpdates request; make sure that only one bot instance is running

**Comment by xmatthias** [comment]
You must use a seperate telegram bot - as documented in the [multiple bot documentation](https://www.freqtrade.io/en/latest/advanced-setup/#running-multiple-instances-of-freqtrade).

Not sure what made you think this would not apply when using docker ...

**Comment by guizmoau** [comment]
yes I do have a different configuration for each bot id but the chat id is the same...

**Comment by guizmoau** [comment]
https://api.telegram.org/bot<botID>/getUpdates

give me the same chat_id

**Comment by xmatthias** [comment]
then you should verify you've got the correct configuration used for each bot.

The chatID will not change - the bot-token will.
If you use the same bot-token for more than one freqtrade bot - you get the above error.

If you can't figure out what's wrong, best register a new telegram bot and try that - so you're sure you've got no "other" system using that bot (which could also be something you don't even remember having running).

**Comment by guizmoau** [comment]
Ok that's clear.

I don't know where it came from that error but know everything is fine after rebooting all docker on the machine

**Comment by yul** [comment]
is there a way to control both bots in the same UI?

I'am using docker, mapped bot1 to 8080, bot2 to 8081 with docker-compose.yaml.
Then I am using nginx to reverse proxy 8080 to port 80 and 8081 to port 81 so they can be used from outside.

I can access localhost:8080 and localhost:8081 from the server terminal
I can access both bots via external_ip:80 and external_ip:81 from outside and see their FreqUIs separately.

However, I cannot add bot2 to bot1's FreqUI, if I try to add external_ip:81 to bot on external_ip:80 (or vise-versa), I just get error "API Url required" and suggestion to check ping URL (which works fine) and CORS documentation, which mentions that it is necessary to run multiple bots on same instance, but doesn't explain how to configure this. 

I've tried putting 

```CORS_origins: ["http://localhost:8080", "http://localhost:8081"] ```

in config.json for both bots, it doesn't change anything.

**Comment by xmatthias** [comment]
Please don't highjack or revive issues that are 2 years old. 
The documentation contains (in my opinion clear) instructions on how cors should be configured (including a screenshot showing WHAT should go into the cors field).

If there's doubts about this - please open a new issue - where we can try to identify what's unclear to you.
